lib/rubocop/cop/mixin/comments_help.rb

```diff
@@ -7,15 +7,9 @@ module CommentsHelp
       include VisibilityHelp
 
       def source_range_with_comment(node)
-        begin_pos, end_pos =
-          if node.def_type?
-            start_node = find_visibility_start(node) || node
-            end_node = find_visibility_end(node) || node
-            [begin_pos_with_comment(start_node),
-             end_position_for(end_node) + 1]
-          else
-            [begin_pos_with_comment(node), end_position_for(node)]
-          end
+        begin_pos = begin_pos_with_comment(node)
+        end_pos = end_position_for(node)
+        end_pos += 1 if node.def_type?
```

Comment:
As mentioned, this method was incorrectly getting a range for not just the `def`, but all `def`s within the same visibility group. I've fixed it to just return the `def` and its preceding comments.